### PR TITLE
virtctl: Separate terminal logic for linux and windows

### LINKED ISSUE #3876: virtctl in Windows 10 throws "Make raw terminal failed: The handle is invalid?" error
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
 /kind bug
> /kind enhancement

**What happened**:

Running virtctl CLI command in Microsoft Windows 10 through Cygwin/Mingw64 CLI returns "Make raw terminal failed: The handle is invalid?" error. The work-around solution exists by using "winpty" in front of the command, but the limitation remains.

**What you expected to happen**:

virtctl command works correctly in Microsoft Windows 10

**How to reproduce it (as minimally and precisely as possible)**:

1. Download virtctl CLI for Windows
2. Set the Environment Variable path for virtctl
3. Open Mingw64 CLI/Cygwin. 
4. Connect to Openshift cluster with CLI command. oc get vm or oc get vmi to get the VM to connect to
5. virtctl console VM_NAME
6. Notice the error message "Make raw terminal failed: The handle is invalid?"

**Anything else we need to know?**:

**Environment**: Microsoft Windows 10
- KubeVirt version (use `virtctl version`):
- Kubernetes version (use `kubectl version`):
- VM or VMI specifications: OCP 4.4.3 cluster
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release): 
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### LINKED ISSUE #8094: virtctl: Native ssh client does not work on Windows 10
**What happened**:

When trying to establish a ssh connection with the native ssh client in virtctl on Windows 10 the connection fails with the following error:

`The handle is invalid.`

**What you expected to happen**:

The ssh connection is established succesfully.

**How to reproduce it (as minimally and precisely as possible)**:

- `virtctl.exe ssh --local-ssh=false user@vm`

**Additional context**:

The same error occurs in CMD, PowerShell and Git Bash.

When using `--local-ssh=true` the connection is established succesfully. (port-forwarding works)

**Environment**:
- KubeVirt version (use `virtctl version`):
```
Client Version: version.Info{GitVersion:"v0.54.0-441+7057484f2c65d0", GitCommit:"7057484f2c65d0df051dd0a7774f23fe3fe374d5", GitTreeState:"clean", BuildDate:"2022-07-13T00:41:39Z", GoVersion:"go1.17.8", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{GitVersion:"v0.49.0-163-gb0553b0f5", GitCommit:"b0553b0f5e5c7c091e7107382073bba7d9be3f43", GitTreeState:"clean", BuildDate:"2022-05-19T18:19:30Z", GoVersion:"go1.16.12", Compiler:"gc", Platform:"linux/amd64"}
```
- Kubernetes version (use `kubectl version`): N/A
- VM or VMI specifications: N/A
- Cloud provider or hardware configuration: N/A
- OS (e.g. from /etc/os-release): Windows 10 21H2
- Kernel (e.g. `uname -a`): N/A
- Install tools: N/A
- Others: N/A

### PR DESCRIPTION
**Pull request by 0xFelix**
**What this PR does / why we need it**:

This separates the logic to initialize the terminal on linux and
windows when using the native ssh client of virtctl. The correct console
modes for the ssh client needed to work are set on windows.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #8094 
Fixes #3876

**Release note**:

```release-note
virtctl's native SSH client is now useable in the Windows console without workarounds
```

### CODE REVIEW

File: pkg/virtctl/ssh/terminal_windows.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	// Enable wanted modes
+	newModeIn |= (windows.ENABLE_WINDOW_INPUT | windows.ENABLE_VIRTUAL_TERMINAL_INPUT)
+	newModeOut := modeOut | windows.ENABLE_VIRTUAL_TERMINAL_PROCESSING |
+		windows.DISABLE_NEWLINE_AUTO_RETURN
+
+	if err := windows.SetConsoleMode(handleIn, newModeIn); err != nil {
+		return nil, err
+	}
+	if err := windows.SetConsoleMode(handleOut, newModeOut); err != nil {
+		return nil, err
```
Comment (rmohr) [inline pkg/virtctl/ssh/terminal_windows.go:44]: Should we try to restore the first step if we fail on the second one?

**Review by rmohr — COMMENTED** [review]
One question. Looks great!

**Comment by rmohr** [comment]
/lgtm
/approve

/hold

Looks great. I have no windows at hand.
I guess you checked it, and if you did feel free to remove the hold :)

**Comment by 0xFelix** [comment]
Works great on CMD, PowerShell and Windows Terminal! :)

/unhold

**Pull request merged**